packages/drivers/routerlicious-driver/src/deltaStorageService.ts

```diff
@@ -61,11 +62,19 @@ export class DocumentDeltaStorageService implements IDocumentDeltaStorageService
 
         this.logtailSha = undefined;
         if (opsFromLogTail.length > 0) {
-            const messages = opsFromLogTail.filter((op) =>
-                op.sequenceNumber >= from,
-            );
-            if (messages.length > 0) {
-                return { messages, partialResult: true };
+            try {
+                const messages = opsFromLogTail.filter((op, i) => {
+                    if (i > 0 && op.sequenceNumber !== opsFromLogTail[i - 1].sequenceNumber + 1) {
```

Comment:
It might be helpful to add some comments here to explain this logic. Maybe it was just me but I am a bit confused with the value of `i`, what should be on `opsFromLogTail[i-1]` and why we add +1 to its sequence number when doing the comparison

Reply:
This is to check if the seq no in opsFromLogTail are continuous. op is the same as opsFromLogTail[i]. So the check here is making sure that every opsFromLogTail[i] is one more than opsFromLogTail[i-1]. If not, we throw and fallback to query deltas from the server

Reply:
Ah! That makes sense. Thanks for the clarification!